### PR TITLE
PEP 686: Grammar fixes

### PR DESCRIPTION
**Pull request by methane**

### CODE REVIEW

**Comment by hugovk** [comment]
I'm very wary of Grammarly, it can make writing worse:

- [chronicle.com/blogs/linguafranca/2012/08/01/these-cards-always-lie](https://www.chronicle.com/blogs/linguafranca/2012/08/01/these-cards-always-lie/)
- [economist.com/johnson/2012/08/01/automated-grammar-checking](https://www.economist.com/johnson/2012/08/01/automated-grammar-checking)
- [economist.com/johnson/2012/08/07/grammarly-revisited](https://www.economist.com/johnson/2012/08/07/grammarly-revisited)
- [the-digital-reader.com/2017/05/04/im-breaking-grammarly](https://the-digital-reader.com/2017/05/04/im-breaking-grammarly/)
- [slate.com/technology/2018/02/grammarly-wont-make-your-writing-better.html](https://slate.com/technology/2018/02/grammarly-wont-make-your-writing-better.html)

And a number of these changes are plain wrong.

File: pep-0686.rst [hunk 1]
```diff
@@ -13,32 +13,33 @@ Post-History: `18-Mar-2022 <https://discuss.python.org/t/14435>`__
 Abstract
 ========
 
-This PEP proposes making :pep:`UTF-8 mode <540>` on by default.
+This PEP proposes to enable :pep:`UTF-8 mode <540>` by default.
```
Comment (hugovk) [inline pep-0686.rst:16]:
```suggestion
This PEP proposes enabling :pep:`UTF-8 mode <540>` by default.
```

File: pep-0686.rst [hunk 2]
```diff
@@ -13,32 +13,33 @@ Post-History: `18-Mar-2022 <https://discuss.python.org/t/14435>`__
...
+files, stdio, and pipes.
 
 
 Motivation
 ==========
 
-UTF-8 becomes de-facto standard text encoding.
+UTF-8 becomes de facto standard text encoding.
 
 * Default encoding of Python source files is UTF-8.
```
Comment (hugovk) [inline pep-0686.rst:27]:
```suggestion
* The default encoding of Python source files is UTF-8.
```

File: pep-0686.rst [hunk 3]
```diff
@@ -13,32 +13,33 @@ Post-History: `18-Mar-2022 <https://discuss.python.org/t/14435>`__
...
-* JSON, TOML, YAML uses UTF-8.
-* Most text editors including VS Code and Windows notepad use UTF-8 by
+* JSON, TOML, YAML use UTF-8.
+* Most text editors, including VS Code and Windows notepad use UTF-8 by
```
Comment (hugovk) [inline pep-0686.rst:29]:
```suggestion
* Most text editors, including Visual Studio Code and Windows Notepad use UTF-8 by
```

File: pep-0686.rst [hunk 4]
```diff
@@ -13,32 +13,33 @@ Post-History: `18-Mar-2022 <https://discuss.python.org/t/14435>`__
...
   default.
-* Most websites and text data on the internet uses UTF-8.
-* And many other popular programming languages including node.js, Go, Rust,
+* Most websites and text data on the internet use UTF-8.
+* And many other popular programming languages, including node.js, Go, Rust,
```
Comment (hugovk) [inline pep-0686.rst:32]:
```suggestion
* And many other popular programming languages, including Node.js, Go, Rust,
```

File: pep-0686.rst [hunk 5]
```diff
@@ -13,32 +13,33 @@ Post-History: `18-Mar-2022 <https://discuss.python.org/t/14435>`__
...
   and Java uses UTF-8 by default.
 
 Changing the default encoding to UTF-8 makes Python easier to interoperate
```
Comment (hugovk) [inline pep-0686.rst:35]:
```suggestion
Changing the default encoding to UTF-8 makes it easier for Python to interoperate
```

File: pep-0686.rst [hunk 6]
```diff
@@ -13,32 +13,33 @@ Post-History: `18-Mar-2022 <https://discuss.python.org/t/14435>`__
...
 with them.
 
 Additionally, many Python developers using Unix forget that the default
-encoding is platform dependant. They omit to specify ``encoding="utf-8"`` when
-they read text files encoded in UTF-8 (e.g. JSON, TOML, Markdown, and Python
-source files). Inconsistent default encoding caused many bugs.
+encoding is a platform dependent.
```
Comment (hugovk) [inline pep-0686.rst:39]:
```suggestion
encoding is platform dependent.
```

File: pep-0686.rst [hunk 7]
```diff
@@ -62,27 +64,29 @@ This behavior is inconsistent with the :pep:`597` motivation.
...
-``UnicodeError``, mojibake, or even silent data corruption. So this change
-should be announced very loudly.
+``UnicodeError``, mojibake, or even silent data corruption.
+So this change should be announced loudly.
 
 To resolve this backward incompatibility, users can do:
 
 * Disable UTF-8 mode.
-* Use ``EncodingWarning`` to find where the default encoding is used and use
+* Use ``EncodingWarning`` to find where the default encoding used and use
```
Comment (hugovk) [inline pep-0686.rst:89]:
```suggestion
* Use ``EncodingWarning`` to find where the default encoding is used and use
```

File: pep-0686.rst [hunk 8]
```diff
@@ -110,22 +114,24 @@ Rejected Alternative
...
+But there are many cases that the default encoding is used for reading/writing
+only ASCII text.
+Additionally, such warnings are not useful for non-cross platform applications
+run on Unix.
 
-So forcing users to specify the ``encoding`` option everywhere is too painful.
+So forcing users to specify the ``encoding`` everywhere is too painful.
 
-Java also rejected this idea in `JEP 400`_.
+Java also rejected this idea in the `JEP 400`_.
```
Comment (hugovk) [inline pep-0686.rst:126]:
```suggestion
Java also rejected this idea in `JEP 400`_.
```

**Review by hugovk — COMMENTED** [review]
A few suggestions. Some of these undo Grammarly, some are new :)

**Comment by CAM-Gerlach** [comment]
Considering their tremendous marketing hype, while I was naturally very skeptical and didn't expect much, they still managed to be rather disappointing. It only catches a fraction of obvious, unambiguous grammar errors, never mind more complex issues; in some cases, the fix doesn't actually improve things, and in others, it changed perfectly valid (and even good-quality) sentences into ones with objective grammar errors. In all, the net result is only a marginal improvement

If you want to automatically fix the various grammar issues with this PEP, I'd suggest turning to CAMmerly instead, which uses a deep learning, neural network-trained copyediting engine tuned specifically on PEPs, all available for free to any PEP author on request—no need to pay for premium service.

**Pull request merged**